tests/utilities/nidaqmx_measurement/__init__.py

```diff
@@ -0,0 +1,87 @@
+import pathlib
+from typing import Iterable, Sequence, Tuple
+
+import nidaqmx
+
+import ni_measurementlink_service as nims
+
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "NIDAQmxMeasurement.serviceconfig",
+    version="0.1.0.0",
+    ui_file_paths=[
+        service_directory,
+    ],
+)
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration("pin_names", nims.DataType.PinArray1D, ["Pin1"])
+@measurement_service.configuration("multi_session", nims.DataType.Boolean, False)
+@measurement_service.output("session_names", nims.DataType.StringArray1D)
+@measurement_service.output("resource_names", nims.DataType.StringArray1D)
+@measurement_service.output("channel_lists", nims.DataType.StringArray1D)
+@measurement_service.output("connected_channels", nims.DataType.StringArray1D)
+@measurement_service.output("voltage_values", nims.DataType.DoubleArray1D)
+def measure(
+    pin_names: Iterable[str],
+    multi_session: bool,
+) -> Tuple[Iterable[str], Iterable[str], Iterable[str], Iterable[str], Iterable[float]]:
+    """NI-DAQmx MeasurementLink test service."""
+    if multi_session:
+        with measurement_service.context.reserve_sessions(pin_names) as reservation:
+            with reservation.create_nidaqmx_tasks() as session_infos:
+                connections = reservation.get_nidaqmx_connections(pin_names)
+                assert all([session_info.session is not None for session_info in session_infos])
+                voltage_values = _read_voltage_values(session_infos)
+
+                return (
+                    [session_info.session_name for session_info in session_infos],
+                    [session_info.resource_name for session_info in session_infos],
+                    [session_info.channel_list for session_info in session_infos],
+                    [connection.channel_name for connection in connections],
+                    voltage_values,
+                )
+
+    else:
+        with measurement_service.context.reserve_session(pin_names) as reservation:
+            with reservation.create_nidaqmx_task() as session_info:
+                connection = reservation.get_nidaqmx_connection(list(pin_names)[0])
+                assert session_info.session is not None
+                voltage_values = _read_voltage_values([session_info])
+
+                return (
+                    [session_info.resource_name],
+                    [session_info.resource_name],
+                    [session_info.channel_list],
+                    [connection.channel_name],
+                    voltage_values,
+                )
+
+
+def _read_voltage_values(
+    session_infos: Sequence[nims.session_management.TypedSessionInformation[nidaqmx.Task]],
+) -> Iterable[float]:
+    sample_rate = 1000.0
+    number_of_samples = 2
+
+    voltage_values = []
+    for session_info in session_infos:
```

Comment:
As with NI-Digital, wrapping the entire multi-session measurement in one for-loop is a mistake because it prevents parallelizing across sessions. Call task.start(), task.read(), and task.stop() in separate for-loops. 

Yes, task.start() and task.stop() are new. Explicitly starting and stopping the tasks allows you to run multiple measurements in parallel.